pkg/apis/storagemigration/validation/validation.go

```diff
@@ -47,16 +58,8 @@ func ValidateStorageVersionMigrationUpdate(newSVMBundle, oldSVMBundle *storagemi
 	allErrs := ValidateStorageVersionMigration(newSVMBundle)
 	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newSVMBundle.ObjectMeta, &oldSVMBundle.ObjectMeta, field.NewPath("metadata"))...)
 
-	// prevent changes to the group, version and resource
-	if newSVMBundle.Spec.Resource.Group != oldSVMBundle.Spec.Resource.Group {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("group"), newSVMBundle.Spec.Resource.Group, "field is immutable"))
-	}
-	if newSVMBundle.Spec.Resource.Version != oldSVMBundle.Spec.Resource.Version {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("version"), newSVMBundle.Spec.Resource.Version, "field is immutable"))
-	}
-	if newSVMBundle.Spec.Resource.Resource != oldSVMBundle.Spec.Resource.Resource {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("resource"), newSVMBundle.Spec.Resource.Resource, "field is immutable"))
-	}
+	// prevent changes to the spec
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldSVMBundle.Spec, newSVMBundle.Spec, field.NewPath("spec"))...)
```

Comment:
So the error message is correct:

```suggestion
	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSVMBundle.Spec, oldSVMBundle.Spec, field.NewPath("spec"))...)
```